src/bridge/views/Login.js

```diff
@@ -0,0 +1,131 @@
+import { Nothing, Just } from 'folktale/maybe';
+import React from 'react';
+import * as azimuth from 'azimuth-js';
+
+import { H1, Row, Col } from '../components/Base';
+import HorizontalSelector from '../components/HorizontalSelector';
+
+import Ticket from './Login/Ticket';
+import Mnemonic from './Login/Mnemonic';
+import Advanced from './Login/Advanced';
+
+import * as need from '../lib/need';
+import { compose } from '../lib/lib';
+import { ROUTE_NAMES } from '../lib/routeNames';
+
+import { withHistory } from '../store/history';
+import { withNetwork } from '../store/network';
+import { withWallet } from '../store/wallet';
+import { withPointCursor } from '../store/pointCursor';
+
+const TABS = {
+  TICKET: Symbol('TICKET'),
+  MNEMONIC: Symbol('MNEMONIC'),
+  ADVANCED: Symbol('ADVANCED'),
+};
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentTab: TABS.TICKET,
+      wallet: Nothing(),
+    };
+
+    this.handleTabChange = this.handleTabChange.bind(this);
+    this.continue = this.continue.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.setNetworkType(this.props.networkType);
+    // we expect pointCursor to not be set yet
+    this.props.setPointCursor(Nothing());
+  }
+
+  handleTabChange(currentTab) {
+    this.setState({ currentTab });
+  }
+
+  async continue() {
+    const wallet = need.wallet(this.props);
+    const contracts = need.contracts(this.props);
+
+    // if no point cursor set by login logic, try to deduce it
+    let deduced = Nothing();
+    if (Nothing.hasInstance(this.props.pointCursor)) {
+      const owned = await azimuth.azimuth.getOwnedPoints(
+        contracts,
+        wallet.address
+      );
+      if (owned.length === 1) {
+        deduced = Just(owned[0]);
+      } else if (owned.length === 0) {
+        const canOwn = await azimuth.azimuth.getTransferringFor(
+          contracts,
+          wallet.address
+        );
+        if (canOwn.length === 1) {
+          deduced = Just(canOwn[0]);
+        }
+      }
+    }
+
+    if (Just.hasInstance(deduced)) {
+      this.props.setPointCursor(deduced);
+      this.props.history.popAndPush(ROUTE_NAMES.POINT_HOME);
+    } else if (Just.hasInstance(this.props.pointCursor)) {
+      this.props.history.popAndPush(ROUTE_NAMES.POINT_HOME);
+    } else {
+      //TODO to new overview (maybe should be merged into point home?)
+      this.props.history.popAndPush(ROUTE_NAMES.SHIPS);
+    }
+  }
+
+  render() {
+    const tabOptions = [
```

Comment:
move this outside of the component to where TABS is?

Reply:
nitpick: can we use `key` instead of `value` for the tab key?

Reply:
I just mimicked what was done in other cases for `HorizontalSelector`. But `key` does also work. I'm fine changing this, but can you explain why you prefer it?

Reply:
Wait no I didn't reload properly. Changing this actually does break `HorizontalSelector`, which checks options for `.value` instead of `.key`.

Reply:
I was suggesting to also change horizontal selector - if we don't want to change it that's cool. the semantics are wrong, since that's a key (not a value) is all